keras_hub/src/models/qwen3_moe/qwen3_moe_backbone.py

```diff
@@ -0,0 +1,366 @@
+            projections in the attention layer.
+        hidden_dim: int. The size of the transformer hidden state at the end of
+            each transformer layer.
+        intermediate_dim: int. The output dimension of the first Dense layer in
+            the feedforward network for each transformer.
+        moe_intermediate_dim: int. The intermediate dimension for each expert
+            in the MoE feedforward network.
+        num_experts: int. The number of experts in each MoE layer.
+        top_k: int. The number of top experts to select for each token in the
+            MoE layer.
+        head_dim: int. The size of each attention head.
+        layer_norm_epsilon: float. The epsilon value used for every layer norm
+            in the transformer model.
+        dropout: float. Dropout probability for the transformer encoder.
+        use_sliding_window_attention: bool. Whether to use sliding local window
+            attention. Defaults to False.
+        sliding_window_size: int. Size of the sliding local window. Defaults to
+            4096.
+        max_sequence_length: int. The maximum sequence length supported by the
+            model. Defaults to 4096.
+        dtype: str or `keras.mixed_precision.DTypePolicy`. The dtype to use for
+            the model's computations and weights. Note that some computations,
+            such as softmax and layer normalization, will always be done at
+            float32 precision regardless of dtype.
+
+    Example:
+    ```python
+    input_data = {
+        "token_ids": np.ones(shape=(1, 12), dtype="int32"),
+        "padding_mask": np.array([[1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0]]),
+    }
+
+    # Pretrained Qwen MoE decoder.
+    model = keras_hub.models.Qwen3MoeBackbone.from_preset("qwen_moe_a2_7b")
+    model(input_data)
+
+    # Randomly initialized Qwen MoE decoder with custom config.
+    model = keras_hub.models.Qwen3MoeBackbone(
+        vocabulary_size=151936,
+        num_layers=28,
+        num_query_heads=16,
+        num_key_value_heads=8,
+        hidden_dim=2048,
+        intermediate_dim=4096,
+        moe_intermediate_dim=128,
+        num_experts=60,
+        top_k=4,
+        head_dim=128,
+        max_sequence_length=4096,
+    )
+    model(input_data)
+    """
+
+    def __init__(
+        self,
+        vocabulary_size,
+        num_layers,
+        num_query_heads,
+        num_key_value_heads,
+        head_dim,
+        hidden_dim,
+        intermediate_dim,
+        moe_intermediate_dim,
+        num_experts,
+        top_k=4,
+        norm_top_k_prob=False,
+        decoder_sparse_step=1,
+        rope_max_wavelength=10000,
+        rope_scaling_factor=1.0,
+        layer_norm_epsilon=1e-6,
+        dropout=0,
+        dtype=None,
+        tie_word_embeddings=False,
+        use_sliding_window_attention=False,
+        sliding_window_size=32768,
+        output_router_logits=False,
+        router_aux_loss_coefficient=0.001,
+        mlp_only_layers=[],
```

Comment:
Fine to have something like this for the toplevel, but let's pass something more direct to each decoder layer (so we don't need to pass the index down). Make sure to document if we keep it.